mlflow/langchain/__init__.py

```diff
@@ -359,7 +366,7 @@ def load_retriever(persist_directory):
         **model_data_kwargs,
     )
 
-    if Version(langchain.__version__) >= Version("0.0.311"):
+    if Version(langchain.__version__) >= Version("0.0.311") and not resources:
```

Comment:
I think this good but what do you think of doing some sort of union operation between detected databricks dependencies and the resources passed in. 

Pros:
1. Users using langchain have not used resources before. So if for example they want to add retrievers, it might be easier for them just to add that and we infer the rest. 
2. If they forget to add something we can still try to cover up their mistakes.
3. The detect here expands the sugar syntax (.*) of the UC functions so that every single function is listed. Users need to remember to add every single function to the resources.yaml if they use the sugar syntax. This can also be fixed with documentation changes.

Cons:
1. Differs from pyfunc model behaviour so can cause confusion

Reply:
I am not sure if its a good idea to merge the resources in a nested level. Also imagine, they provide the resources X and assume that Y would be inferred but it wouldn't be the case. Its rather be a break glass approach where users can provide the entire list similar to Pyfunc IMO. It would be very difficult for the users on what is inferred v/s not IMO.

Reply:
Ya thats a good point